resources/extension.rb

```diff
@@ -16,18 +15,41 @@
 # limitations under the License.
 #
 
-property :database,    String, required: true
-property :extension,   String, name_property: true
-property :old_version, String
+property :extension,        String, name_property: true
+property :old_version,      String
+property :source_directory, String
+property :version,          String, default: '--1.0'
+
+# Connection prefernces
+property :user,     String, default: 'postgres'
+property :database, String, required: true
+property :host,     [String, nil]
+property :port,     Integer, default: 5432
 
 action :create do
-  create_query = "CREATE EXTENSION IF NOT EXISTS \"#{new_resource.extension}\""
-  create_query << " FROM \"#{new_resource.old_version}\"" if property_is_set?(:old_version)
+  extension_path = ::File.join(new_resource.source_directory, "#{new_resource.extension}#{new_resource.version}.sql")
+  cmd = %(psql -f "#{extension_path}" -d test_1 -U postgres --port 5432)
```

Comment:
Is it possible to use the full path of the psql command ? I know a bunch of admin who create system wide alias to force some options (who said RedHat maintainers ?)
I don't know if they are willing to do it on psql command... But ideally, i would try to use the real binay instead of a possible alias.